libledger/Ledger.cpp

```diff
@@ -351,11 +351,33 @@ void Ledger::initDBConfig(ptree const& pt)
     {
         m_param->mutableStorageParam().maxRetry = 100;
     }
+    /// set state db related param
+    m_param->mutableStateParam().type = pt.get<std::string>("state.type", "storage");
+
+    // read db config from config eg:mysqlip mysqlport and so on
+    m_param->mutableStorageParam().dbtype = pt.get<std::string>("storage.dbtype", "mysql");
+    m_param->mutableStorageParam().dbip = pt.get<std::string>("storage.dbip", "127.0.0.1");
+    m_param->mutableStorageParam().dbport = pt.get<int>("storage.dbport", 3306);
+    m_param->mutableStorageParam().dbusername = pt.get<std::string>("storage.dbusername", "");
+    m_param->mutableStorageParam().dbpasswd = pt.get<std::string>("storage.dbpasswd", "");
+    m_param->mutableStorageParam().dbname = pt.get<std::string>("storage.dbname", "");
+    m_param->mutableStorageParam().dbcharset = pt.get<std::string>("storage.dbcharset", "utf8");

```

Comment:
https://stackoverflow.com/questions/30074492/what-is-the-difference-between-utf8mb4-and-utf8-charsets-in-mysql
is this ok to set utf8?